AutomatticTracks/src/main/java/com/automattic/android/tracks/crashlogging/internal/SentryCrashLogging.kt

```diff
@@ -134,6 +140,46 @@ internal class SentryCrashLogging constructor(
         sentryWrapper.captureEvent(event)
     }
 
+    override fun sendJavaScriptReport(
+        jsException: JsException,
+        callback: JsExceptionCallback
+    ) {
+        val frames = jsException.stackTrace.map {
+            SentryStackFrame().apply {
+                this.filename = it.fileName
+                this.function = it.function
+                this.lineno = it.lineNumber
+                this.colno = it.colNumber
+                this.isInApp = true
+            }
+        }.toMutableList()
+
+        val sentryException = SentryException().apply {
+            this.type = jsException.type
+            this.value = jsException.message
+            this.module = "javascript"
+            this.stacktrace = SentryStackTrace().apply { this.frames = frames }
+            this.mechanism = Mechanism().apply {
+                this.isHandled = jsException.isHandled
+                this.type = jsException.handledBy
+            }
+        }
+
+        val event = SentryEvent().apply {
+            this.message = Message().apply { this.message = message }
+            this.level = SentryLevel.FATAL
+            this.platform = "javascript"
+            this.appendTags(jsException.tags)
+            this.exceptions = mutableListOf(sentryException)
+        }
+
+        Sentry.configureScope { scope ->
+            scope.setContexts("react_native_context", jsException.context)
+        }
+        sentryWrapper.captureEvent(event)
+        callback.onReportSent(true)
```

Comment:
@jhnstn I'm not sure if this will guarantee that the event has been sent at the time the callback is triggered. Did you have the chance to confirm it?

Reply:
eeek, I'm not sure how to do that. I noticed you use some deferred queue and then call flush in the iOS tracker. 
I'm guessing we could use [coroutines](https://developer.android.com/kotlin/coroutines) here, but I'm not entirely sure why you use a deferred queue. Is the flush call blocking until the milliseconds run out? 

I did dig into the Sentry SDK to try to figure out what flush does, but I ran into this comment: 

```
Flushes events queued up to the current hub. Not implemented yet.
```


I can keep digging but I'm still unsure why we need this level of inspection. Do we expect `captureEvent` to fail often? can we catch the conditions where we think it will fail before calling?  Is there a concern that we might be silencing exceptions due to malformed expectation data?

Reply:
> Is the flush call blocking until the milliseconds run out?

Yes. It tries to flush all queued Sentry events and finishes either when all events are sent or hitting the timeout ([reference](https://github.com/Automattic/Automattic-Tracks-iOS/blob/d3694e8a352f8cac3f3c26f73125614408c01abc/Sources/Remote%20Logging/Crash%20Logging/CrashLogging.swift#L244)). I haven't checked if the Android implementation works the same way, but if the events are being queued, we'd need to find a way to ensure they are sent. As I shared [here](https://github.com/WordPress/gutenberg/pull/59221#discussion_r1508682107), the main purpose of the callback is to ensure the event is sent to Sentry before letting the app crash. 

> Do we expect `captureEvent` to fail often? can we catch the conditions where we think it will fail before calling? Is there a concern that we might be silencing exceptions due to malformed expectation data?

We don't need to check for failures when sending the event. As I shared above, the callback's purpose is mainly to ensure that the event is sent.

@jhnstn This can be tested by forcing an unhandled exception and checking in Sentry if it gets reported before the app crashes. I'm currently testing this in installable builds.

Reply:
I've just validated that the Sentry event is sent before the crash 🎊. Not sure if it works because on Android the crash only affects the editor screen (i.e. the app doesn't fully crash) or if Sentry events are not being queued. In any case, I confirm this approach works.